server/core/build.rs

```diff
@@ -1,4 +1,40 @@
+use std::path::{Path, PathBuf};
+
+/// Work out where the workspace dir is
+fn workspace_dir() -> PathBuf {
+    let output = std::process::Command::new(env!("CARGO"))
+        .arg("locate-project")
+        .arg("--workspace")
+        .arg("--message-format=plain")
+        .output()
+        .unwrap()
+        .stdout;
+    let cargo_path = Path::new(std::str::from_utf8(&output).unwrap().trim());
+    cargo_path.parent().unwrap().to_path_buf()
+}
+
+/// Determine the git rev, we do this here so it's only actually run and checked once at build time.
+fn determine_git_rev() -> Option<String> {
```

Comment:
Not everyone will build out of git, so this will break things. We sohuld use the kanidm version instead here as it contains a commit hash on devel builds.

Reply:
https://github.com/kanidm/kanidm/blob/master/libs/profiles/src/lib.rs#L86

Reply:
I'd say rather than a direct commit rev, we just sha2 the current KANIDM_PKG_VERSION instead and use KANIDM_PKG_HASH. Potentially we should probably truncate that hash so that it's not the full 16 bytes or whatever of hex.

Reply:
> Not everyone will build out of git, so this will break things. We sohuld use the kanidm version instead here as it contains a commit hash on devel builds.

It wouldn't break anything - it returns `None` if the git repo's not found and falls back to CARGO_PKG_VERSION in the function that grabs it. 

I'd forgotten about KANIDM_PKG_VERSION, I'll use the hash of that instead (I'd use it as-is but it has a space in it, which can't be used :smile:)